p2p/src/network/scheduler/p2p_network_scheduler_reducer.rs

```diff
@@ -64,148 +132,480 @@ impl P2pNetworkSchedulerState {
+                let dispatcher = state_context.into_dispatcher();
+
+                match kind {
+                    SelectKind::Stream(peer_id, stream_id)
+                        if keep_connection_with_unknown_stream() =>
+                    {
+                        warn!(meta.time(); summary="select error for stream", addr = display(addr), peer_id = display(peer_id));
+                        // just close the stream
+                        dispatcher.push(P2pNetworkYamuxAction::OutgoingData {
+                            addr: *addr,
+                            stream_id: *stream_id,
+                            data: Data::default(),
+                            flags: YamuxFlags::RST,
+                        });
+                        dispatcher.push(P2pNetworkSchedulerAction::PruneStream {
+                            peer_id: *peer_id,
+                            stream_id: *stream_id,
+                        });
+                    }
+                    _ => {
+                        dispatcher.push(P2pNetworkSchedulerAction::Error {
+                            addr: *addr,
+                            error: P2pNetworkConnectionError::SelectError,
+                        });
+                    }
+                }
+
+                dispatcher.push(P2pNetworkSchedulerEffectfulAction::SelectError {
+                    addr: *addr,
+                    kind: *kind,
+                    error: error.to_owned(),
+                });
+
+                Ok(())
             }
             P2pNetworkSchedulerAction::YamuxDidInit {
                 addr,
                 message_size_limit,
-                ..
+                peer_id,
             } => {
-                if let Some(cn) = self.connections.get_mut(addr) {
-                    if let Some(P2pNetworkConnectionMuxState::Yamux(yamux)) = &mut cn.mux {
-                        yamux.init = true;
-                        yamux.message_size_limit = *message_size_limit;
-                    }
+                let Some(cn) = scheduler_state.connections.get_mut(addr) else {
+                    return Ok(());
+                };
+                if let Some(P2pNetworkConnectionMuxState::Yamux(yamux)) = &mut cn.mux {
+                    yamux.init = true;
+                    yamux.message_size_limit = *message_size_limit;
+                }
+
+                let incoming = cn.incoming;
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let peer_id = *peer_id;
+                let addr = *addr;
+
+                if incoming {
+                    dispatcher.push(P2pConnectionIncomingAction::Libp2pReceived { peer_id });
+                } else {
+                    dispatcher.push(P2pConnectionOutgoingAction::FinalizeSuccess { peer_id });
+                }
+
+                // for each negotiated yamux conenction open a new outgoing RPC stream
```

Comment:
Now that I see this comment I wonder if we should do this after receiving the identity message from the peer (which should include Rpc in the list of supported protocols).